### PR TITLE
New sample: Display points using clustering feature reduction

### PR DESCRIPTION
**Pull request by williambohrmann3**
# Description

Adds new clustering sample to `Layers` category.

## Type of change

- New sample implementation

## Platforms tested on

- [x] WPF .NET 6
- [x] WPF Framework
- [x] WinUI
- [x] MAUI WinUI
- [x] MAUI Android
- [x] MAUI iOS
- [x] MAUI MacCatalyst

## Checklist

- [x] Self-review of changes
- [x] All changes work as expected on all affected platforms
- [x] There are no warnings related to changes
- [x] Code is commented and follows .NET conventions and standards
- [x] Codemaid and XAML styler extensions have been run on every changed file
- [x] No unrelated changes have been made to any other code or project files
- [x] Screenshots are correct size and display in description tab (800 x 600 on Windows, MAUI mobile platforms use the MAUI Windows screenshot)

### CODE REVIEW

**Comment by williambohrmann3** [comment]
popup viewer not displaying right on MAUI iOS or MacCatalyst
when clicking away, popup viewer width/height size changes

https://github.com/Esri/arcgis-maps-sdk-dotnet-samples/assets/118313312/919a8ecf-bf2e-4ee7-950b-9cb040b72a86

File: src/MAUI/Maui.Samples/Samples/Layers/DisplayPointsUsingClusteringFeatureReduction/readme.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+![](displaypointsusingclusteringfeaturereduction.jpg)
+
+## Use case
+
+Feature clustering can be used to dynamically aggregate groups of points that are within proximity of each other in order to represent each group with a single symbol. Such grouping allows you to see patterns in the data that are difficult to visualize when a layer contains hundreds or thousands of points that overlap and cover each other.
+
+## How to use the sample
+
+Pan and zoom the map to view how clustering is dynamically updated. Toggle clustering off to view the original point features that make up the clustered elements. When clustering is On, you can tap on a clustered geoelement to view aggregated information and summary statistics for that cluster. When clustering is toggled off and you tap on the original feature you get access to information about individual power plant features.
```
Comment (duffh) [inline src/MAUI/Maui.Samples/Samples/Layers/DisplayPointsUsingClusteringFeatureReduction/readme.md:13]:
```suggestion
Pan and zoom the map to view how clustering is dynamically updated. Toggle clustering off to view the original point features that make up the clustered elements. When clustering is on, you can tap on a clustered geoelement to view aggregated information and summary statistics for that cluster. When clustering is toggled off and you tap on the original feature you get access to information about individual power plant features.
```

Does "On" need a capital? Off does not have one.

File: src/MAUI/Maui.Samples/Samples/Layers/DisplayPointsUsingClusteringFeatureReduction/DisplayPointsUsingClusteringFeatureReduction.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            PopupBackground.IsVisible = true;
+        }
+
+        // Enable clustering feature reduction if the checkbox has been checked, disable otherwise.
+        private void CheckBox_CheckChanged(object sender, CheckedChangedEventArgs e)
+        {
+            // This event is raised when sample is initially loaded when layer is null.
+            if (_layer == null) return;
+
+            _layer.FeatureReduction.IsEnabled = (bool)(sender as CheckBox).IsChecked;
```
Comment (duffh) [inline src/MAUI/Maui.Samples/Samples/Layers/DisplayPointsUsingClusteringFeatureReduction/DisplayPointsUsingClusteringFeatureReduction.xaml.cs:76]:
```suggestion
            _layer.FeatureReduction.IsEnabled = (sender as CheckBox).IsChecked;
```

Cast doesn't seem to be required.

**Review by duffh — COMMENTED** [review]
Minor comments

File: src/MAUI/Maui.Samples/Samples/Layers/DisplayPointsUsingClusteringFeatureReduction/DisplayPointsUsingClusteringFeatureReduction.xaml.cs (hunk 2 above)
Comment (williambohrmann3) [reply src/MAUI/Maui.Samples/Samples/Layers/DisplayPointsUsingClusteringFeatureReduction/DisplayPointsUsingClusteringFeatureReduction.xaml.cs]: This is not required for MAUI, but is still required for WinUI and WPF.

**Comment by pMaske** [comment]
@williambohrmann3 Did popup layout fix with latest toolkit changes?

**Comment by williambohrmann3** [comment]
> @williambohrmann3 Did popup layout fix with latest toolkit changes?

@pMaske no, still able to repro on iOS and MacCatalyst w/ daily3961

**Comment by williambohrmann3** [comment]
iOS behavior:

https://github.com/Esri/arcgis-maps-sdk-dotnet-samples/assets/118313312/040b6354-f96b-4a71-8c26-e05295f7690f

Notice that popup width/height is corrected upon foregrounding app.

**Comment by dotMorten** [comment]
Currently I'd suggest putting popupviewer content in a popupmodal.

**Comment by williambohrmann3** [comment]
Popupviewer working fine in landscape mode on Android and iOS simulators I tested. Text fits inside 250x250 popupviewer, control displays fine no cutoff. Can always change to navigation stack if we decide to for UIX purpose or discover another poupviewer bug but it doesn't seem necessary for the time being.

**Review by duffh — APPROVED** [review]

**Pull request merged**